Apps/W1/Shopify/app/src/Payments/Codeunits/ShpfyPayments.Codeunit.al

```diff
@@ -203,4 +203,47 @@ codeunit 30169 "Shpfy Payments"
             end;
         end;
     end;
+
+    internal procedure UpdateDisputeStatus()
+    var
+        PaymentTransaction: Record "Shpfy Payment Transaction";
+        DisputeToken: JsonToken;
+        JResponse: JsonToken;
+        Url: Text;
+        UrlTxt: Label 'shopify_payments/disputes/%1.json', Locked = true;
+    begin
+        PaymentTransaction.SetRange(Type, PaymentTransaction.Type::Dispute);
+        PaymentTransaction.SetFilter("Dispute Status", '<>%1&<>%2', PaymentTransaction."Dispute Status"::Won, PaymentTransaction."Dispute Status"::Lost);
+
+        if PaymentTransaction.FindSet() then
+            repeat
+                Url := CommunicationMgt.CreateWebRequestURL(StrSubstNo(UrlTxt, PaymentTransaction.Id));
```

Comment:
Does this work? Is PaymentTransaction.Id = Dispute.Id?

Reply:
Good catch. Fixing asap

Reply:
@onbuyuka 

This was the actual reason why I previously picked up all Disputes every time. The dispute ids are linked to an Order rather then a payment transaction.

I restructured the code, added a Dispute table. Added code that will import data into the Dispute table with SinceId, and another function that will update existing disputes.

Then , instead of presenting the data on the transaction line (considering it's linked to an order), I added an action that will open all disputes related to the same order. Also added the same action on the Order page.